docs/Modules/multichain.mdx

```diff
@@ -1,22 +1,136 @@
+This module acts as an extension of the [ECDSA Module](/Modules/ecdsa). The added benefit of this module is you now have the ability to sign once and authorize any number of userOps across multiple blockchain networks.
+
+### What is the Multichain Validation Module? 🤔
+
+This module allows **Externally Owned Accounts (EOAs)** to authorize and sign user operations (UserOps) for Biconomy Smart Accounts on Multiple chains with a single ECDSA signature. 
+
+**Benefits**
+- significantly improves UX for deploying and setting up Smart Accounts on several chains 
+- significantly reduces user friction for multichain operations.
+
+:::tip
+You can also use this module as the **default module** just like the ECDSA Validation Module, as it has all its capabilities for sending ops on the particular chain.
+:::
+
+#### Key Functions 🔐
+
+- **ECDSA Signature Scheme**: Utilizes the ECDSA secp256r1 curve for secure signing.
+- **Flexible Signer Authentication**: Supports various signer solutions like Privy, Fireblocks, Arcana Auth, Capsule, Lit protocol, Turnkey, Web3Auth, Particle, Magic, Portal, etc.
+- **EIP-1271 Compliance**: Allowing Smart Accounts to sign Ethereum messages for logging into dApps.
+- **Sign once, Execute on multiple chains**
+
+#### Use Cases 🌟
+
+Enables use cases which require several actions to be authorized for several chains with just one signature required from user.
+
+1. Multichain intent for example exit all my native tokens for USDC
+2. Allows a dApp to require just one signature from their user to configure & deploy smart accounts on multiple chains and delegate certain actions with permission via session keys on all those chains
+3. Cross-chain state sync actions e.g transfer ownership of a smart account, change ENS, similar actions on a protocol that are deployed on multiple chains
+4. Onboard a user on x chains by deploying their smart account and some action like registering in a raffle 
+
+### How does it work? 🤔
+
+1. Collecting the set of userOps for signing
+2. Calculate userOpHash
+3. Every userOpHash is now a leaf of a Merkle Tree.
+4. Instead of signing individual hashes, the user signs the Merkle Root.
+
+Now this signed Merkle Root serves as on-chain evidence, proving any userOp within the tree is authorized. With this, Users can authorize any number of userOps with just one signature over the Merkle Tree Root.
+
+
+## Smart Contract Deep Dive 🛠️
+
+You have the option to transmit a multi-chain signature, enabling the sending of various user operations on distinct chains. This signature is then validated by the module. Alternatively, if a plain ECDSA signature (65 bytes) is sent, the module interprets it as a standard ECDSA validation.
+
+#### User Operation Validation (`validateUserOp`) 🛡️
+
+```solidity
```

Comment:
```suggestion
```typescript
```

Docuaurus doesn't properly handle solidity and lacks ofcode colors. Using Typescript here would improve it

Reply:
We _could_ add it, since Prism already supports it. https://docusaurus.io/docs/markdown-features/code-blocks#supported-languages

https://prismjs.com/#supported-languages

Reply:
@livingrockrises in the docusaurus.config.js you can add the following 
 prism: {
        theme: lightCodeTheme,
        darkTheme: darkCodeTheme,
        **additionalLanguages: ['solidity']**
      },
 and use ```sol ``` it works. also can we shorten this method by moving some params to function line.